File: Core/automation/jsr223/javascript/core/000_startup_delay.js
'use strict';

var Thread = Java.type("java.lang.Thread");
var log = Java.type("org.slf4j.LoggerFactory").getLogger("jsr223.javascript.core.startup_delay");

log.info("Checking for initialized context");

while (true) {
    try {
        scriptExtension.importPreset("RuleSupport");
        if (automationManager != null) {
            break;
        }
    }
    catch(err) {
        log.info("Context not initialized yet... waiting 10s before checking again");
        Thread.sleep(10000);
    }
}

log.info("Context initialized... waiting 30s before allowing scripts to load");
Thread.sleep(30000);
log.info("Complete");

File: Core/automation/jsr223/python/core/000_startup_delay.py
"""
This script is designed to delay the loading of other scripts until openHAB is
completely started.
"""

from time import sleep

from org.slf4j import Logger, LoggerFactory

log = LoggerFactory.getLogger("jsr223.jython.core.startup_delay")
log.info("Checking for initialized context")

while True:
    try:
        scriptExtension.importPreset("RuleSupport")
        if automationManager is not None:
            break
    except:
        log.info("Context not initialized yet... waiting 10s before checking again")
        sleep(10)

log.info("Context initialized... waiting 30s before allowing scripts to load")
sleep(30)
log.info("Complete")
